hits_ingest.py

```diff
@@ -5,88 +5,102 @@
-if doIngestCalibs: sqlite3 database registry of ingested calibration products
-                   created in calibrepo
-if doProcessCcd: outputrepo/visit populated with subdirectories containing the
-                 usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR)
-
-BASH EQUIVALENT
-if doIngest:
+ingest: repo populated with *links* to files in datadir, organized by date
+        sqlite3 database registry of ingested images also created in repo
+ingestCalibs: calibrepo populated with *links* to files in datadir,
+              organized by date (bias/zero and flat images only)
+              sqlite3 database registry of ingested calibration products
+              created in calibrepo
+        **NOTE: this does not ingest any defects or fringes!**
+        You must do this manually... e.g.,
+        $ cd calibrepo
+        $ ingestCalibs.py ../repo --calib . --calibType defect --validity 999
+          ../HiTS/MasterCals/defects/2014-12-05/*fits
+processCcd: processedrepo/visit populated with subdirectories containing the
+            usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR)
+
+BASH EQUIVALENTS
+ingest:
     $ ingestImagesDecam.py repo --filetype raw --mode link datafiles
-if doIngestCalibs:
-    $ ingestCalibs.py repo --calib calibrepo --validity 999 datafiles
-if doProcessCcd:
+ingestCalibs:
+    $ ingestCalibs.py repo --calib calibrepo --mode=link --validity 999 datafiles
+processCcd:
     $ cd calibrepo
-    $ processCcd.py ../repo --id visit=visit ccdnum=ccdnum 
-            --output ../outputrepo -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
-            --config calibrate.doAstrometry=False calibrate.doPhotoCal=False
-            --no-versions
+    $ processCcd.py repo --id visit=visit ccdnum=ccdnum
+            --output processedrepo --calib calibrepo
+            -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
+            -C processccd_config.py
+            --config calibrate.doAstrometry=True calibrate.doPhotoCal=True
+    ** note that to run from bash, 'processccd_config.py' must exist
+diffIm:
+    $ imageDifference.py processedrepo --id visit=diffimvisit ccdnum=ccdnum
+            --templateId visit=templatevisit --output diffimrepo
+            -C diffim_config.py
+    ** note that to run from bash, 'diffim_config.py' must exist
 '''
-# edit values below as desired
-doIngest = False
-doIngestCalibs = False
-doProcessCcd = True
-visit = '410877' # used for ProcessCcd only
-ccdnum = '25'    # used for ProcessCcd only
-repo = 'ingested/'
-calibrepo = 'calibingested/'
-outputrepo = 'processed/'
-# edit values above as desired
-
-if ((doIngest and doIngestCalibs) or (doIngest and doProcessCcd) or 
-    (doIngestCalibs and doProcessCcd)):
-    raise RuntimeError('You can only have one task True at a time')
-elif not doIngest and not doIngestCalibs and not doProcessCcd:
-    raise RuntimeError('Nothing to do, every task is set to False')
-
-if len(sys.argv) < 2:
+# ~~  edit values below as desired  ~~ #
```

Comment:
I haven't implemented a yaml config file, but I did clean up the input significantly.